routing/pathfind.go

```diff
@@ -108,7 +109,8 @@ type finalHopParams struct {
 // dependencies.
 func newRoute(sourceVertex route.Vertex,
 	pathEdges []*channeldb.CachedEdgePolicy, currentHeight uint32,
-	finalHop finalHopParams) (*route.Route, error) {
+	finalHop finalHopParams, blindedPath *sphinx.BlindedPath) (
```

Comment:
NIT: Maybe add a little bit of explanation about the `blindedPath` to the function documentation, i.e. that `blindedPath` is used to backfill the fields of the `pathEdges` for the part that is considered the blinded path. The commit message makes it very clear but from the function signature alone it may be confusing to understand if `pathEdges` already contain "blinded hops" or not?